app/models/manageiq/providers/azure/refresh_helper_methods.rb

```diff
@@ -105,56 +105,78 @@ def network_routers
   end
 
   # Create the necessary service classes and lock down their api-version
-  # strings using the config/settings.yml from the provider repo. The
-  # "to_s" call for the version strings puts the date in the format
-  # that we need, i.e. "YYYY-MM-DD".
+  # strings using the config/settings.yml from the provider repo.
+
+  def cached_resource_provider_service(config)
+    @cached_resource_provider_service ||= resource_provider_service(config)
+  end
+
+  # If the api-version string set in settings.yml is invalid, a warning
+  # will be issued, and it will default to the most recent valid string.
   #
+  def valid_api_version(config, service, name)
+    valid_api_versions = cached_resource_provider_service(config).list_api_versions(service.provider)
+    config_api_version = Settings.ems.ems_azure.api_versions[name]
+
+    if valid_api_versions.include?(config_api_version)
+      config_api_version
+    else
+      valid_version_string = valid_api_versions.first
+
+      message = "Invalid api-version setting of '#{config_api_version}' for " \
+        "#{service.provider}/#{service.service_name} for EMS #{@ems.name}; " \
+        "using '#{valid_version_string}' instead."
+
+      _log.warn(message)
```

Comment:
Is it possible to detect this at the time it's set in the settings?  I can show you where to do it...my only concern is if it makes a call out to the provider (can't tell from this code if it does or not).

Reply:
It does make one extra call internally to get that list.

Reply:
I would like to see how to do that @Fryguy